**SOCIAL RESEARCH: Final Project** 

**NAME:** Letizia

**SURNAME:** Lanza

**ID:** VR502321

**PART 1:** Data Exploration and Data Wrangling

The first thing to succesfully carry out some interesting exercises, is to import some libraries as we have seen in class such as:

In [15]:
import os 
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns

The aim of the following command is being able to read data from a CSV file and load it into a Pandas DataFrame.

In [16]:
obesity_df = pd.read_csv("ObesityDataSet_raw_and_data_sinthetic.csv")
obesity_df

Age  Gender    Height      Weight        CALC FAVC  FCVC  NCP  \
0     21.000000  Female  1.620000   64.000000          no   no   2.0  3.0   
1     21.000000  Female  1.520000   56.000000   Sometimes   no   3.0  3.0   
2     23.000000    Male  1.800000   77.000000  Frequently   no   2.0  3.0   
3     27.000000    Male  1.800000   87.000000  Frequently   no   3.0  3.0   
4     22.000000    Male  1.780000   89.800000   Sometimes   no   2.0  1.0   
...         ...     ...       ...         ...         ...  ...   ...  ...   
2106  20.976842  Female  1.710730  131.408528   Sometimes  yes   3.0  3.0   
2107  21.982942  Female  1.748584  133.742943   Sometimes  yes   3.0  3.0   
2108  22.524036  Female  1.752206  133.689352   Sometimes  yes   3.0  3.0   
2109  24.361936  Female  1.739450  133.346641   Sometimes  yes   3.0  3.0   
2110  23.664709  Female  1.738836  133.472641   Sometimes  yes   3.0  3.0   

      SCC SMOKE      CH2O family_history_with_overweight       FAF       TUE  \
0      no    no  2.000000                            yes  0.000000  1.000000   
1     yes   yes  3.000000                            yes  3.000000  0.000000   
2      no    no  2.000000                            yes  2.000000  1.000000   
3      no    no  2.000000                             no  2.000000  0.000000   
4      no    no  2.000000                             no  0.000000  0.000000   
...   ...   ...       ...                            ...       ...       ...   
2106   no    no  1.728139                            yes  1.676269  0.906247   
2107   no    no  2.005130                            yes  1.341390  0.599270   
2108   no    no  2.054193                            yes  1.414209  0.646288   
2109   no    no  2.852339                            yes  1.139107  0.586035   
2110   no    no  2.863513                            yes  1.026452  0.714137   

           CAEC                 MTRANS           NObeyesdad  
0     Sometimes  Public_Transportation        Normal_Weight  
1     Sometimes  Public_Transportation        Normal_Weight  
2     Sometimes  Public_Transportation        Normal_Weight  
3     Sometimes                Walking   Overweight_Level_I  
4     Sometimes  Public_Transportation  Overweight_Level_II  
...         ...                    ...                  ...  
2106  Sometimes  Public_Transportation     Obesity_Type_III  
2107  Sometimes  Public_Transportation     Obesity_Type_III  
2108  Sometimes  Public_Transportation     Obesity_Type_III  
2109  Sometimes  Public_Transportation     Obesity_Type_III  
2110  Sometimes  Public_Transportation     Obesity_Type_III  

[2111 rows x 17 columns]

As it can be seen the dataset has 2111 rows and 17 columns.

It is also important to explain what the 17 columns refer to: 
- Age: feature, continuous, "Age"
- Gender: feature, categorical, "Gender 
- Height: feature, continuous 
- Weight: feature, continuous 
- CALC: feature, categorical, "how often do you drink alcohol?"
- FAVC: feature, binary, "do you eat high caloric food frequently?"
- FCVC: feature, integer, "do you usually eat vegetables in your meals?"
- NCP: feature, continuous, "how many meals do you have daily?"
- SCC: feature, binary, "do you monitor the calories you eat daily?"
- SMOKE: feature, binary, "do you smoke?"
- CH2O: feature, continuous, "how much water do you drink daily?"
- family_history_with_overweight: feature, binary, "has a family member suffered or suffers from overweight?"
- FAF: feature, continuous, "how often do you have physical activity?"
- TUE: feature, integer, "how much time do you use technological devices such as cell phone, videogames, television, computer and others?"
- CAEC: feature, categorical, "do you eat any food between meals?"
- MTRANS: feature, categorical, "which transportation do you usually use?"
- NObeyesdad: target, categorical, "obesity level" 

More specifically, *"features"*, in a Social Research environment, are attributes or variables that are considered and analyzed to better understand social phenomena, relationships, and human behavior. Also, when a variable is *"continuous"* it means that it is a characteristic or attribute that can take on an infinite number of values within a specified range. These variables are measurable on a continuous scale and are not restricted to discrete or categorical values. In fact, when features are *"categorical"*, it signifies that they correspond variables that take discrete values and can be organized into distinct categories or levels rather than on a continuous scale. These categories can be represented by labels or nominal values that do not imply intrinsic ordering. Again, when features are *"binary"*, it indicates that they exemplify variables that take only two distinct values or categories. These two categories can be delineated by labels such as "0" and "1," "yes" and "no," "present" and "absent," "male" and "female," "success" and "failure," and so on. Finally, when features are of type *"integer"* it denotes that they represent numeric variables that take only integer values, that is, numbers without fractions or decimals. These variables are typically measurable on an integer scale and cannot take non - integer values.

The purpose of the following codeds is to round to one decimal place those values that have multilple decimal places. This is done for columns such as "Age", "Height", "Weight", "CH2O", "FAF" and "TUE". 

In [17]:
obesity_df['Age'] = obesity_df['Age'].round(1)
obesity_df['Height'] = obesity_df['Height'].round(1)
obesity_df['Weight'] = obesity_df['Weight'].round(1)
obesity_df['CH2O'] = obesity_df['CH2O'].round(1)
obesity_df['FAF'] = obesity_df['FAF'].round(1)
obesity_df['TUE'] = obesity_df['TUE'].round(1)
obesity_df

Age  Gender  Height  Weight        CALC FAVC  FCVC  NCP  SCC SMOKE  \
0     21.0  Female     1.6    64.0          no   no   2.0  3.0   no    no   
1     21.0  Female     1.5    56.0   Sometimes   no   3.0  3.0  yes   yes   
2     23.0    Male     1.8    77.0  Frequently   no   2.0  3.0   no    no   
3     27.0    Male     1.8    87.0  Frequently   no   3.0  3.0   no    no   
4     22.0    Male     1.8    89.8   Sometimes   no   2.0  1.0   no    no   
...    ...     ...     ...     ...         ...  ...   ...  ...  ...   ...   
2106  21.0  Female     1.7   131.4   Sometimes  yes   3.0  3.0   no    no   
2107  22.0  Female     1.7   133.7   Sometimes  yes   3.0  3.0   no    no   
2108  22.5  Female     1.8   133.7   Sometimes  yes   3.0  3.0   no    no   
2109  24.4  Female     1.7   133.3   Sometimes  yes   3.0  3.0   no    no   
2110  23.7  Female     1.7   133.5   Sometimes  yes   3.0  3.0   no    no   

      CH2O family_history_with_overweight  FAF  TUE       CAEC  \
0      2.0                            yes  0.0  1.0  Sometimes   
1      3.0                            yes  3.0  0.0  Sometimes   
2      2.0                            yes  2.0  1.0  Sometimes   
3      2.0                             no  2.0  0.0  Sometimes   
4      2.0                             no  0.0  0.0  Sometimes   
...    ...                            ...  ...  ...        ...   
2106   1.7                            yes  1.7  0.9  Sometimes   
2107   2.0                            yes  1.3  0.6  Sometimes   
2108   2.1                            yes  1.4  0.6  Sometimes   
2109   2.9                            yes  1.1  0.6  Sometimes   
2110   2.9                            yes  1.0  0.7  Sometimes   

                     MTRANS           NObeyesdad  
0     Public_Transportation        Normal_Weight  
1     Public_Transportation        Normal_Weight  
2     Public_Transportation        Normal_Weight  
3                   Walking   Overweight_Level_I  
4     Public_Transportation  Overweight_Level_II  
...                     ...                  ...  
2106  Public_Transportation     Obesity_Type_III  
2107  Public_Transportation     Obesity_Type_III  
2108  Public_Transportation     Obesity_Type_III  
2109  Public_Transportation     Obesity_Type_III  
2110  Public_Transportation     Obesity_Type_III  

[2111 rows x 17 columns]

The DataFrame.info() function prints information about the DataFrame such as number of columns, column labels, column data types, memory usage, range index and the number of cells in each column (non-null values):

In [18]:
obesity_df.info(verbose = True, show_counts = True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2111 entries, 0 to 2110
Data columns (total 17 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Age                             2111 non-null   float64
 1   Gender                          2111 non-null   object 
 2   Height                          2111 non-null   float64
 3   Weight                          2111 non-null   float64
 4   CALC                            2111 non-null   object 
 5   FAVC                            2111 non-null   object 
 6   FCVC                            2111 non-null   float64
 7   NCP                             2111 non-null   float64
 8   SCC                             2111 non-null   object 
 9   SMOKE                           2111 non-null   object 
 10  CH2O                            2111 non-null   float64
 11  family_history_with_overweight  2111 non-null   object 
 12  FAF                             21

In particular, in this case it was necessary to add the verbose = True and the show_counts = True functions to get a broader and more detailed overview of the dataset. Moreover, the verbose = True function stands for controlling the verbosity of the output and when it is set to 'True', it provides some additional information like column names and data types. The show_count = True function, also, shows the number of non-null values for each column alongside the total row count.

Hence, as it can be seen there are 17 columns with a Rangeindex (rows) of 2111 (as seen before). Moreover, the Non - Null Count column returns the number of non - null values in the relative column of the DataFrame, as said before. There are also two types of data types such as object (generic data) and float (real numbers that can have a fractional part). Another information that can be read is the memory usage of 280.5+ KB.

The DataFrame.describe() function gives descriptive information about the dataset as it can be seen below:

In [19]:
obesity_df.describe()

Age       Height       Weight         FCVC          NCP  \
count  2111.00000  2111.000000  2111.000000  2111.000000  2111.000000   
mean     24.31279     1.701232    86.585978     2.419043     2.685628   
std       6.34572     0.099470    26.191165     0.533927     0.778039   
min      14.00000     1.400000    39.000000     1.000000     1.000000   
25%      19.90000     1.600000    65.450000     2.000000     2.658738   
50%      22.80000     1.700000    83.000000     2.385502     3.000000   
75%      26.00000     1.800000   107.450000     3.000000     3.000000   
max      61.00000     2.000000   173.000000     3.000000     4.000000   

              CH2O          FAF          TUE  
count  2111.000000  2111.000000  2111.000000  
mean      2.007816     1.010279     0.658077  
std       0.614717     0.851329     0.609296  
min       1.000000     0.000000     0.000000  
25%       1.600000     0.100000     0.000000  
50%       2.000000     1.000000     0.600000  
75%       2.500000     1.700000     1.000000  
max       3.000000     3.000000     2.000000

Glossary:

count: the number of non-null values in each column
mean: the mean (average) value of each numerical column
sdt: the standard deviation of each numerical column
min: the minimum value in each numerical column
25%: the value at the 25th percentile (first quartile) of each numerical column
50%: the value at the 50th percentile (median or second quartile) of each numerical column
75%: the value at the 75th percentile (third quartile) of each numerical column
max: the maximum value in each numerical column

The DataFrame.describe().T function is used to transpose index and columns of the data frame as follow:

In [20]:
obesity_df.describe().T

count       mean        std   min        25%        50%     75%  \
Age     2111.0  24.312790   6.345720  14.0  19.900000  22.800000   26.00   
Height  2111.0   1.701232   0.099470   1.4   1.600000   1.700000    1.80   
Weight  2111.0  86.585978  26.191165  39.0  65.450000  83.000000  107.45   
FCVC    2111.0   2.419043   0.533927   1.0   2.000000   2.385502    3.00   
NCP     2111.0   2.685628   0.778039   1.0   2.658738   3.000000    3.00   
CH2O    2111.0   2.007816   0.614717   1.0   1.600000   2.000000    2.50   
FAF     2111.0   1.010279   0.851329   0.0   0.100000   1.000000    1.70   
TUE     2111.0   0.658077   0.609296   0.0   0.000000   0.600000    1.00   

          max  
Age      61.0  
Height    2.0  
Weight  173.0  
FCVC      3.0  
NCP       4.0  
CH2O      3.0  
FAF       3.0  
TUE       2.0

The DataFrame.head() function displays the first 5 rows of the DataFrame by default as below:

In [21]:
obesity_df.head()

Age  Gender  Height  Weight        CALC FAVC  FCVC  NCP  SCC SMOKE  CH2O  \
0  21.0  Female     1.6    64.0          no   no   2.0  3.0   no    no   2.0   
1  21.0  Female     1.5    56.0   Sometimes   no   3.0  3.0  yes   yes   3.0   
2  23.0    Male     1.8    77.0  Frequently   no   2.0  3.0   no    no   2.0   
3  27.0    Male     1.8    87.0  Frequently   no   3.0  3.0   no    no   2.0   
4  22.0    Male     1.8    89.8   Sometimes   no   2.0  1.0   no    no   2.0   

  family_history_with_overweight  FAF  TUE       CAEC                 MTRANS  \
0                            yes  0.0  1.0  Sometimes  Public_Transportation   
1                            yes  3.0  0.0  Sometimes  Public_Transportation   
2                            yes  2.0  1.0  Sometimes  Public_Transportation   
3                             no  2.0  0.0  Sometimes                Walking   
4                             no  0.0  0.0  Sometimes  Public_Transportation   

            NObeyesdad  
0        Normal_Weight  
1        Normal_Weight  
2        Normal_Weight  
3   Overweight_Level_I  
4  Overweight_Level_II

The DataFrame.tail() function returns the last 5 rows of the dataframe by default as can be seen:

In [22]:
obesity_df.tail()

Age  Gender  Height  Weight       CALC FAVC  FCVC  NCP SCC SMOKE  CH2O  \
2106  21.0  Female     1.7   131.4  Sometimes  yes   3.0  3.0  no    no   1.7   
2107  22.0  Female     1.7   133.7  Sometimes  yes   3.0  3.0  no    no   2.0   
2108  22.5  Female     1.8   133.7  Sometimes  yes   3.0  3.0  no    no   2.1   
2109  24.4  Female     1.7   133.3  Sometimes  yes   3.0  3.0  no    no   2.9   
2110  23.7  Female     1.7   133.5  Sometimes  yes   3.0  3.0  no    no   2.9   

     family_history_with_overweight  FAF  TUE       CAEC  \
2106                            yes  1.7  0.9  Sometimes   
2107                            yes  1.3  0.6  Sometimes   
2108                            yes  1.4  0.6  Sometimes   
2109                            yes  1.1  0.6  Sometimes   
2110                            yes  1.0  0.7  Sometimes   

                     MTRANS        NObeyesdad  
2106  Public_Transportation  Obesity_Type_III  
2107  Public_Transportation  Obesity_Type_III  
2108  Public_Transportation  Obesity_Type_III  
2109  Public_Transportation  Obesity_Type_III  
2110  Public_Transportation  Obesity_Type_III

The DataFrame.columns function is an attribute that provides access to the column labels of a DataFrame:

In [23]:
obesity_df.columns

Index(['Age', 'Gender', 'Height', 'Weight', 'CALC', 'FAVC', 'FCVC', 'NCP',
       'SCC', 'SMOKE', 'CH2O', 'family_history_with_overweight', 'FAF', 'TUE',
       'CAEC', 'MTRANS', 'NObeyesdad'],
      dtype='object')

The DataFrame.max() function is used to find the maximum value in the DataFrame:

In [24]:
obesity_df.max()

Age                                              61.0
Gender                                           Male
Height                                            2.0
Weight                                          173.0
CALC                                               no
FAVC                                              yes
FCVC                                              3.0
NCP                                               4.0
SCC                                               yes
SMOKE                                             yes
CH2O                                              3.0
family_history_with_overweight                    yes
FAF                                               3.0
TUE                                               2.0
CAEC                                               no
MTRANS                                        Walking
NObeyesdad                        Overweight_Level_II
dtype: object

The DataFrame.min() function is used to find the minimum value in the DataFrame:

In [25]:
obesity_df.min()

Age                                              14.0
Gender                                         Female
Height                                            1.4
Weight                                           39.0
CALC                                           Always
FAVC                                               no
FCVC                                              1.0
NCP                                               1.0
SCC                                                no
SMOKE                                              no
CH2O                                              1.0
family_history_with_overweight                     no
FAF                                               0.0
TUE                                               0.0
CAEC                                           Always
MTRANS                                     Automobile
NObeyesdad                        Insufficient_Weight
dtype: object

For checking if there are some null values the DataFrame.isnull() function can be used:

In [26]:
obesity_df.isnull()

Age  Gender  Height  Weight   CALC   FAVC   FCVC    NCP    SCC  SMOKE  \
0     False   False   False   False  False  False  False  False  False  False   
1     False   False   False   False  False  False  False  False  False  False   
2     False   False   False   False  False  False  False  False  False  False   
3     False   False   False   False  False  False  False  False  False  False   
4     False   False   False   False  False  False  False  False  False  False   
...     ...     ...     ...     ...    ...    ...    ...    ...    ...    ...   
2106  False   False   False   False  False  False  False  False  False  False   
2107  False   False   False   False  False  False  False  False  False  False   
2108  False   False   False   False  False  False  False  False  False  False   
2109  False   False   False   False  False  False  False  False  False  False   
2110  False   False   False   False  False  False  False  False  False  False   

       CH2O  family_history_with_overweight    FAF    TUE   CAEC  MTRANS  \
0     False                           False  False  False  False   False   
1     False                           False  False  False  False   False   
2     False                           False  False  False  False   False   
3     False                           False  False  False  False   False   
4     False                           False  False  False  False   False   
...     ...                             ...    ...    ...    ...     ...   
2106  False                           False  False  False  False   False   
2107  False                           False  False  False  False   False   
2108  False                           False  False  False  False   False   
2109  False                           False  False  False  False   False   
2110  False                           False  False  False  False   False   

      NObeyesdad  
0          False  
1          False  
2          False  
3          False  
4          False  
...          ...  
2106       False  
2107       False  
2108       False  
2109       False  
2110       False  

[2111 rows x 17 columns]

Another way to identify how many null values there are in the dataframe is:

In [27]:
obesity_df.isnull().sum()

Age                               0
Gender                            0
Height                            0
Weight                            0
CALC                              0
FAVC                              0
FCVC                              0
NCP                               0
SCC                               0
SMOKE                             0
CH2O                              0
family_history_with_overweight    0
FAF                               0
TUE                               0
CAEC                              0
MTRANS                            0
NObeyesdad                        0
dtype: int64

As evident from the codes above there is no null value. This could be given by: 
- *Data quality:* the data were collected accurately and completely, with no omissions or missing information. This suggests that the dataset is reliable and ready to be used for analysis without having to deal with missing data issues
- *Data pre - processing:* data pre - processing operations were performed to handle any missing values prior to analysis. These operations may include removing rows or columns with missing values, imputing missing values with estimates based on other data, or using more advanced techniques to deal with missing data
- *Data collection methods:* could reflect the effectiveness of the data collection methods used in the survey or social survey. If participants answered all questions without leaving any blanks, this could indicate a good understanding of the instructions by the participants or a research design that minimized the possibility of missing values